abel/transform.py

```diff
@@ -16,322 +16,356 @@
+
+        ``hansenlaw`` 
+            This "recursive algorithm" produces reliable results
+            and is quite fast (~0.1 sec for a 1001x1001 image).
+            It makes no assumptions about the data
+            (apart from cylindrical symmetry). It tends to require that the data
+            is finely sampled for good convergence.
+
+            E. W. Hansen and P.-L. Law "Recursive methods for computing
+            the Abel transform and its inverse"
+            J. Opt. Soc. A*2, 510-520 (1985)
+            http://dx.doi.org/10.1364/JOSAA.2.000510
+
+        ``basex`` * 
+            The "basis set exapansion" algorithm describes the data in terms
+            of gaussian functions, which themselves can be abel transformed
+            analytically. Because the gaussian functions are approximately the 
+            size of each pixel, this method also does not make any assumption 
+            about the shape of the data. This method is one of the de-facto 
+            standards in photoelectron/photoion imaging.
+
+             Dribinski et al, 2002 (Rev. Sci. Instrum. 73, 2634)
+             http://dx.doi.org/10.1063/1.1482156
 
-    abel_transform = {
-        "basex" : basex.basex_transform,
-        "direct" : direct.direct_transform,
-        "hansenlaw" : hansenlaw.hansenlaw_transform,
-        "three_point" : three_point.three_point_transform,
-    }
-
-    verboseprint = print if verbose else lambda *a, **k: None
-
-    if IM.ndim == 1 or np.shape(IM)[0] <= 2:
+        ``direct``
+            This method attempts a direct integration of the Abel
+            transform integral. It makes no assumptions about the data
+            (apart from cylindrical symmetry),
+            but it typically requires fine sampling to converge.
+            Such methods are typically inefficient,
+            but thanks to this Cython implementation (by Roman Yurchuk),
+            this 'direct' method is competitive with the other methods.
+
+        ``three_point`` *
+            The "Three Point" Abel transform method
+            exploits the observation that the value of the Abel inverted data
+            at any radial position r is primarily determined from changes
+            in the projection data in the neighborhood of r.
+            This method is also very efficient
+            once it has generated the basis sets.
+
+            Dasch, 1992 (Applied Optics, Vol 31, No 8, March 1992, Pg 1146-1152).
+
+        ``*``
+            The methods marked with a * indicate methods that generate basis sets.
+            The first time they are run for a new image size,
+            it takes seconds to minutes to generate the basis set.
+            However, this basis set is saved to disk can can be reloaded,
+            meaning that future transforms are performed
+            much more quickly.
+
+        """
+
+        # public class variables
+        self.IM = IM
```

Comment:
I would prefer not to keep the original image if I want to use pickle to store the transformed results.